code/modules/hivemind/mobs.dm

```diff
@@ -0,0 +1,913 @@
+	playsound(src, 'sound/effects/blobattack.ogg', 70, 1)
+	addtimer(CALLBACK(src, .proc/release_passenger), 20 SECONDS)
+
+
+
+/mob/living/simple_animal/hostile/hivemind/mechiver/proc/release_passenger(var/safely = FALSE)
+	if(passenger)
+		if(pilot)
+			flick("mechiver-opening", src)
+		else
+			flick("mechiver-opening_wires", src)
+
+		if(istype(passenger, /mob/living/carbon/human))
+			if(!safely) //that was stressful
+				var/mob/living/carbon/human/H = passenger
+				if(!pilot && H.stat != CONSCIOUS) //it's very dangerous
+					destroy_passenger()
+					pilot = TRUE
+					return
+
+				H.hallucination = rand(30, 90)
+		//if mob is dead, we just rebuild it
+		if(passenger.stat == DEAD && !safely)
+			dead_body_restoration(passenger)
+
+		if(passenger) //if passenger still here, then just release him
+			passenger << SPAN_DANGER("[src] released you!")
+			passenger.canmove = TRUE
+			passenger.loc = get_turf(src)
+			passenger = null
+			special_ability_cooldown = world.time + ability_cooldown
+		playsound(src, 'sound/effects/blobattack.ogg', 70, 1)
+
+
+/mob/living/simple_animal/hostile/hivemind/mechiver/proc/dead_body_restoration(mob/living/corpse)
+	var/picked_mob
+	if(passenger.mob_size <= MOB_SMALL && !client && prob(50))
+		picked_mob = pick(/mob/living/simple_animal/hostile/hivemind/stinger, /mob/living/simple_animal/hostile/hivemind/bomber)
+	else
+		if(pilot)
+			if(istype(corpse, /mob/living/carbon/human))
+				picked_mob = /mob/living/simple_animal/hostile/hivemind/himan
+			else if(istype(corpse, /mob/living/silicon/robot))
+				picked_mob = /mob/living/simple_animal/hostile/hivemind/hiborg
+	if(picked_mob)
+		new picked_mob(get_turf(src))
+	else
+		var/mob/living/simple_animal/hostile/hivemind/resurrected/fixed_mob = new(get_turf(src))
+		fixed_mob.absorb_the(corpse)
+	destroy_passenger()
+
+
+/mob/living/simple_animal/hostile/hivemind/mechiver/proc/destroy_passenger()
+	qdel(passenger)
+	passenger = null
+
+
+//we're not forgot to release our victim safely after death
+/mob/living/simple_animal/hostile/hivemind/mechiver/Destroy()
+	release_passenger(TRUE)
+	..()
+
+/mob/living/simple_animal/hostile/hivemind/mechiver/death()
+	release_passenger(TRUE)
+	..()
+	gibs(loc, null, /obj/effect/gibspawner/robot)
+	if(pilot)
+		gibs(loc, null, /obj/effect/gibspawner/human)
+	qdel(src)
+
+
+/////////////////////////////////////PHASER///////////////////////////////////
+//Special ability: Superposition. Phaser exists at four locations. But, actually he vulnerable only at one. Other is just a copies
+//Moves with teleportation only, can stun victim if he land on it
+//Also can hide in closets
+//Can't speak, no malfunctions
+//Appears from dead human body
+//////////////////////////////////////////////////////////////////////////////
+
+/mob/living/simple_animal/hostile/hivemind/phaser
+	name = "phaser"
+	desc = "Crooked man with strange device on it's head. He twitches sometimes and... What are you looking on? Run!"
```

Comment:
A Crooked human with a strange device on its head. It twitches sometimes and... Why are you still looking? Run!